tests/ui/chars_enumerate_for_byte_indices.rs

```diff
@@ -0,0 +1,59 @@
+#![feature(round_char_boundary)]
+#![warn(clippy::chars_enumerate_for_byte_indices)]
+
+trait StrExt {
+    fn use_index(&self, _: usize);
+}
+impl StrExt for str {
+    fn use_index(&self, _: usize) {}
+}
+
+fn bad(prim: &str, string: String) {
+    for (idx, _) in prim.chars().enumerate() {
+        let _ = prim[..idx];
+        //~^ chars_enumerate_for_byte_indices
+        prim.split_at(idx);
+        //~^ chars_enumerate_for_byte_indices
+
+        // This won't panic, but it can still return a wrong substring
+        let _ = prim[..prim.floor_char_boundary(idx)];
+        //~^ chars_enumerate_for_byte_indices
+
+        // can't use #[expect] here because the .fixed file will still have the attribute and create an
+        // unfulfilled expectation, but make sure lint level attributes work on the use expression:
+        #[allow(clippy::chars_enumerate_for_byte_indices)]
+        let _ = prim[..idx];
```

Comment:
this is a fun one, I wonder if that's something that could be fixed in uitest, like removing `#[expect]` attributes in the .fixed file 🤔